### PR TITLE
add amazon library, ail hover case, and ail select search suggestion …

### PR DESCRIPTION
**Pull request by MikeLien**
…case

### CODE REVIEW

**Comment by MikeLien** [comment]
@ShakoHo @askeing @ypwalter r?

File: lib/sikuli/amazon.sikuli/amazon.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        [os.path.join('pics', 'function_under_search_field_moztpeqa_win10.png'), 0, 0],
+    ]
+
+    def wait_for_loaded(self, similarity=0.70):
+        """
+        @param similarity: The similarity of AMAZON_FUNCTION_UNDER_SEARCH_FIELD component. Default: 0.70.
+        """
+        return self._wait_for_loaded(component=Amazon.AMAZON_FUNCTION_UNDER_SEARCH_FIELD, similarity=similarity)
+
+    def wait_for_suggestion_loaded(self, component):
```
Comment (askeing) [inline lib/sikuli/amazon.sikuli/amazon.py]: In your case, the `component` should be rename to `pattern`.
```python
component = app.wait_for_loaded()
...
type('m')
app.wait_for_suggestion_loaded(component)
```

File: lib/sikuli/common.sikuli/common.py [hunk 2]
```diff
@@ -157,6 +165,25 @@ def _wait_for_loaded(self, component, similarity=0.70, timeout=10):
         wait(p, wait_sec)
         return p
 
+    def _wait_component_for_vanished(self, component, timeout=10):
+        """
+        Wait for component loaded. Default timeout is 10 sec, min is 1 sec.
+        @param component: Specify the wait vanished component, which is an sikuli object pattern.
+        @param timeout: Wait timeout second, the min timeout is 1 sec. Default is 10 sec.
+        @param similarity: The pattern comparing similarity, from 0 to 1. Default is 0.70.
```
Comment (askeing) [inline lib/sikuli/common.sikuli/common.py]: Does not have `similarity` parameter here.

File: lib/sikuli/common.sikuli/common.py [hunk 3]
```diff
@@ -157,6 +165,25 @@ def _wait_for_loaded(self, component, similarity=0.70, timeout=10):
         wait(p, wait_sec)
         return p
 
+    def _wait_component_for_vanished(self, component, timeout=10):
```
Comment (askeing) [inline lib/sikuli/common.sikuli/common.py]: the `component` should be rename to `pattern`.

**Pull request merged**